proxy_map.go

```diff
@@ -0,0 +1,1023 @@
+func (m *Map) ExecuteOnEntries(entryProcessor interface{}) ([]types.Entry, error) {
+	if processorData, err := m.validateAndSerialize(entryProcessor); err != nil {
+		return nil, err
+	} else {
+		ch := make(chan []proto.Pair, 1)
+		handler := func(msg *proto.ClientMessage) {
+			ch <- codec.DecodeMapExecuteOnAllKeysResponse(msg)
+		}
+		request := codec.EncodeMapExecuteOnAllKeysRequest(m.name, processorData)
+		if _, err := m.invokeOnRandomTarget(m.ctx, request, handler); err != nil {
+			return nil, err
+		}
+		pairs := <-ch
+		if kvPairs, err := m.convertPairsToEntries(pairs); err != nil {
+			return nil, err
+		} else {
+			return kvPairs, nil
+		}
+	}
+}
+
+// Flush flushes all the local dirty entries.
+func (m *Map) Flush() error {
+	request := codec.EncodeMapFlushRequest(m.name)
+	_, err := m.invokeOnRandomTarget(m.ctx, request, nil)
+	return err
+}
+
+// ForceUnlock releases the lock for the specified key regardless of the lock owner.
+// It always successfully unlocks the key, never blocks, and returns immediately.
+func (m *Map) ForceUnlock(key interface{}) error {
+	if keyData, err := m.validateAndSerialize(key); err != nil {
+		return err
+	} else {
+		refID := m.refIDGenerator.NextID()
+		request := codec.EncodeMapForceUnlockRequest(m.name, keyData, refID)
+		_, err = m.invokeOnKey(m.ctx, request, keyData)
+		return err
+	}
+}
+
+// Get returns the value for the specified key, or nil if this map does not contain this key.
+// Warning:
+// This method returns a clone of original value, modifying the returned value does not change the
+// actual value in the map. One should put modified value back to make changes visible to all nodes.
+func (m *Map) Get(key interface{}) (interface{}, error) {
+	if keyData, err := m.validateAndSerialize(key); err != nil {
+		return nil, err
+	} else {
+		request := codec.EncodeMapGetRequest(m.name, keyData, m.lockID)
+		if response, err := m.invokeOnKey(m.ctx, request, keyData); err != nil {
+			return nil, err
+		} else {
+			return m.convertToObject(codec.DecodeMapGetResponse(response))
+		}
+	}
+}
+
+// GetAll returns the entries for the given keys.
+func (m *Map) GetAll(keys ...interface{}) ([]types.Entry, error) {
+	partitionToKeys := map[int32][]pubserialization.Data{}
+	ps := m.proxy.partitionService
+	for _, key := range keys {
+		if keyData, err := m.validateAndSerialize(key); err != nil {
+			return nil, err
+		} else {
+			partitionKey := ps.GetPartitionID(keyData)
+			arr := partitionToKeys[partitionKey]
+			partitionToKeys[partitionKey] = append(arr, keyData)
+		}
+	}
+	result := make([]types.Entry, 0, len(keys))
+	// create futures
+	f := func(partitionID int32, keys []pubserialization.Data) cb.Future {
+		return m.circuitBreaker.TryContext(m.ctx, func(ctx context.Context) (interface{}, error) {
+			request := codec.EncodeMapGetAllRequest(m.name, keys)
+			inv := m.invokeOnPartitionAsync(request, partitionID)
+			return inv.GetWithTimeout(1 * time.Second)
```

Comment:
Is this one second a leftover ? 
Same goes for putAll as well?

Reply:
`invokeOnPartitionAsync` just creates the invocation, it doesn't wait it to be resolved. But we don't need that anymore, so I've replaced it with invokeOnPartition.

Reply:
I  put back invokeOnPartitionAsync.